velox/dwio/parquet/tests/writer/ParquetWriterTest.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/dwio/parquet/tests/ParquetTestBase.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::common;
+using namespace facebook::velox::dwio::common;
+using namespace facebook::velox::parquet;
+
+class ParquetWriterTest : public ParquetTestBase {
+ protected:
+  std::unique_ptr<RowReader> createRowReaderWithSchema(
+      const std::unique_ptr<Reader> reader,
+      const RowTypePtr& rowType) {
+    auto rowReaderOpts = getReaderOpts(rowType);
+    auto scanSpec = makeScanSpec(rowType);
+    rowReaderOpts.setScanSpec(scanSpec);
+    auto rowReader = reader->createRowReader(rowReaderOpts);
+    return rowReader;
+  };
+};
+
+class SupportedParquetWriterCompressionTest
+    : public ParquetWriterTest,
+      public testing::WithParamInterface<CompressionKind> {
+ public:
+  SupportedParquetWriterCompressionTest() : compressionKind_(GetParam()) {}
+  static std::vector<CompressionKind> getTestParams() {
+    static std::vector<CompressionKind> params = {
+        CompressionKind::CompressionKind_NONE,
+        CompressionKind::CompressionKind_SNAPPY,
+        CompressionKind::CompressionKind_ZSTD,
+        CompressionKind::CompressionKind_LZ4,
+        CompressionKind::CompressionKind_GZIP,
+    };
+    return params;
+  }
+
+ protected:
+  CompressionKind compressionKind_;
+};
+
+TEST_P(SupportedParquetWriterCompressionTest, compression) {
+  auto schema = ROW({"c0", "c1"}, {INTEGER(), DOUBLE()});
+  auto data = makeRowVector({
+      makeFlatVector<int32_t>(10, [](auto row) { return row + 5; }),
```

Comment:
let's write `10'000` values. Make this a constant so that we can use it for the Reader as well.